dnsimple/config.go

```diff
@@ -30,22 +33,22 @@ type Client struct {
 }
 
 // Client returns a new client for accessing DNSimple.
-func (c *Config) Client() (*Client, error) {
-	ts := oauth2.StaticTokenSource(&oauth2.Token{AccessToken: c.Token})
+func (config *Config) Client() (*Client, diag.Diagnostics) {
+	ts := oauth2.StaticTokenSource(&oauth2.Token{AccessToken: config.Token})
 	tc := oauth2.NewClient(context.Background(), ts)
 
 	client := dnsimple.NewClient(tc)
-	client.SetUserAgent(httpclient.TerraformUserAgent(c.terraformVersion))
-	if c.Sandbox {
+	client.SetUserAgent(fmt.Sprintf("HashiCorp Terraform/%s (+https://www.terraform.io) Terraform Plugin SDK/%s", config.terraformVersion, meta.SDKVersionString()))
```

Comment:
FYI: As per the documentation, there is an alternative method of defining the user agent, and it's recommended in case we do not need to keep the same format. https://www.terraform.io/docs/extend/guides/v2-upgrade-guide.html#removal-of-httpclient-package